test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -562,8 +547,22 @@ object Spec {
 
   final class ProvideSomeLayerShared[R0 <: Has[_], -R, +E, +T](private val self: Spec[R, E, T]) extends AnyVal {
     def apply[E1 >: E, R1 <: Has[_]](
-      layer: ZLayer[R0, E1, R1]
+      layer: ZLayer[R0, E1, R1],
+      defExec: ExecutionStrategy = ExecutionStrategy.ParallelN(4)
     )(implicit ev1: R0 with R1 <:< R, ev2: NeedsEnv[R], tagged: Tagged[R1]): Spec[R0, E1, T] =
-      self.provideLayerShared[E1, R0, R0 with R1](ZLayer.identity[R0] ++ layer)
+      self.provideLayerShared[E1, R0, R0 with R1](ZLayer.identity[R0] ++ layer, defExec)
   }
+
+  /**
+   * Applies the function `f` to each element of the `Iterable[A]` and returns
+   * the result in a new `List[B]` using the specified execution strategy.
+   */
+  private def foreachExec[R, E, A, B](
```

Comment:
This could be worth upstreaming to ZIO Core. It provides a nice way of abstracting over execution strategies.

Reply:
I agree, I like it.